system/lib/emmalloc.c

```diff
@@ -225,7 +225,7 @@ an allocation size to the bucket index that should be looked at. The buckets are
   Bucket 62: [100663296, 134217727], range size=33554432
   Bucket 63: 134217728 bytes and larger. */
 static_assert(NUM_FREE_BUCKETS == 64, "Following function is tailored specifically for NUM_FREE_BUCKETS == 64 case");
-static int compute_free_list_bucket(uint32_t allocSize)
+static int compute_free_list_bucket(size_t allocSize)
```

Comment:
The `bucketIndex` computation is likely wrong in 64-bit mode, since `__builtin_clz` will return a value that is much larger than from before, leading to the asserts on lines 233-234 triggering.